site.py

```diff
@@ -0,0 +1,84 @@
+import os
+import subprocess
+import urllib
+import json
+import time
+import random
+
+from flask import Flask, render_template, url_for, request, redirect, send_from_directory
+
+
+app = Flask(__name__)
+
+# Confirm that a provided file has an acceptable extension.
+# In this case, '.csv' is the only allowed extension.
+def allowed_file(filename):
+    return ('.' in filename) and (filename.rsplit('.', 1)[1] in ALLOWED_EXTENSIONS)
+
+# Retrieve the files uploaded by the user, and if they are valid,
+#  store them in a new generated subfolder of tmp/.
+@app.route('/', methods=['GET', 'POST'])
+def upload_file():
+    if request.method == 'POST':
+        metadatafile = request.files['metadatafile']
+        datafile = request.files['datafile']
+        
+        if metadatafile and metadatafile.filename.endswith('.csv') \
+           and datafile and datafile.filename.endswith('.csv'):
+            print "Saving files..."
+            app.config['UPLOAD_FOLDER'] = 'tmp/'+str(random.randint(10000,99999))+'/'
+            while os.path.exists(app.config['UPLOAD_FOLDER']):
+                app.config['UPLOAD_FOLDER'] = 'tmp/'+str(random.randint(10000,99999))+'/'
+            
+            os.mkdir(app.config['UPLOAD_FOLDER'])
+            metadatafile.save(
+                os.path.join(
+                    app.config['UPLOAD_FOLDER'],
+                    metadatafile.filename
+                )
+            )
+            datafile.save(
+                os.path.join(
+                    app.config['UPLOAD_FOLDER'],
+                    datafile.filename
+                )
+            )
+            return redirect(url_for(
+                                'uploaded_file', 
+                                datafilename = datafile.filename, 
+                                metadatafilename = metadatafile.filename
+                            ))
+
+    # If method is not 'POST' (i.e., the user has not uploaded files yet)
+    #  or if files are not accepted, load the form.
+    return render_template('index.html')
+
+# Generate the Docker container from the uploaded files, 
+#  and redirect to the new Harvest instance.
+@app.route('/uploads/<datafilename>/<metadatafilename>')
+def uploaded_file(datafilename, metadatafilename):
```

Comment:
There is no need to expose this as an endpoint. Using file paths as part a URL is not intuitive and prone to error. Secondly, this endpoint could be requested many times which would launch a bunch of containers (even if they fail and exit).

A better approach would be to rely on the UUID that was generated above which is guaranteed unique and URL friendly. In addition, because it points the folder we can store the *state* of the container, so if the URL is requested again the state can be checked rather than blindly running a new container.